x/wasm-storage/keeper/msg_server.go

```diff
@@ -56,19 +59,38 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 
 var _ types.MsgServer = msgServer{}
 
+// StoreDataRequestWasm handles the processing of a store data request for a wasm-storage module.
+// It unzips the Wasm bytecode, and creates a new Wasm instance,
+// checks if a module with the same hash already exists, sets the Wasm instance in the data store,
+// tracks its expiration, emits an event, and returns the hash of the stored Wasm module.
 func (m msgServer) StoreDataRequestWasm(goCtx context.Context, msg *types.MsgStoreDataRequestWasm) (*types.MsgStoreDataRequestWasmResponse, error) {
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	params, err := m.Params.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-	unzipped, err := unzipWasm(msg.Wasm)
+	unzipped, err := unzipWasm(msg.Wasm, params.MaxWasmSize)
 	if err != nil {
 		return nil, err
 	}
-	wasm := types.NewWasm(unzipped, msg.WasmType, ctx.BlockTime())
+
+	wasm := types.NewWasm(unzipped, msg.WasmType, ctx.BlockTime(), ctx.BlockHeight(), params.WasmTTL)
 	if exists, _ := m.DataRequestWasm.Has(ctx, WasmKey(*wasm)); exists {
 		return nil, fmt.Errorf("data Request Wasm with given hash already exists")
 	}
 
-	if err := m.DataRequestWasm.Set(ctx, WasmKey(*wasm), *wasm); err != nil {
+	wasmKey := WasmKey(*wasm)
+	if err := m.DataRequestWasm.Set(ctx, wasmKey, *wasm); err != nil {
+		return nil, err
+	}
+
+	// Track Exp.
+	expKey := collections.Join(wasm.PruneHeight, wasmKey)
```

Comment:
No need for this variable + Fix comment to "Track expiration."

Reply:
[03d50ff](https://github.com/sedaprotocol/seda-chain/pull/266/commits/03d50ff1e022660b4ecdb43672cc2c1f7f1045a9)